src/service-api.ts

```diff
@@ -111,6 +114,48 @@ const plugin: FastifyPluginAsync<GraaspImportZipPluginOptions> = async (fastify,
     return newItems;
   };
 
+  const addFileToZip = async (item: Item, dirPath, archive, member) => {
+    // get item and its related data
+    const itemExtra = item.extra as Extra;
+
+    switch (item.type) {
+      case SERVICE_ITEM_TYPE:
+        // TODO: add uploaded files into zip
+        break;
+      case ItemType.DOCUMENT:
+        archive.append(itemExtra.document?.content, {
+          name: path.join(dirPath, `${item.name}.graasp`),
+        });
+        break;
+      case ItemType.LINK:
+        archive.append(buildTextContent(itemExtra.embeddedLink?.url, ItemType.LINK), {
+          name: path.join(dirPath, `${item.name}.url`),
+        });
+        break;
+      case ItemType.APP:
+        archive.append(buildTextContent(itemExtra.app?.url, ItemType.APP), {
+          name: path.join(dirPath, `${item.name}.url`),
+        });
+        break;
+      case ItemType.FOLDER:
+        // append description
+        // eslint-disable-next-line no-case-declarations
+        const newDirPath = path.join(dirPath, item.name);
+        if (item.description) {
+          archive.append(item.description, {
+            name: path.join(newDirPath, `${item.name}.description.html`),
+          });
+        }
+        // eslint-disable-next-line no-case-declarations
+        const tasks = iTM.createGetChildrenTaskSequence(member, item.id, true);
+        // eslint-disable-next-line no-case-declarations
+        const subItems = await runner.runSingleSequence(tasks);
+        (subItems as Item[]).forEach((subItem) => {
+          addFileToZip(subItem, newDirPath, archive, member);
```

Comment:
This is an async function but you don't await? Is the result okay?

Reply:
I tested myself, it is working well (as the path is independent with each other anyways, so it should work without waiting for the others), so I didn't use await here as it's faster